prime-router/src/main/kotlin/cli/tests/Hl7Tests.kt

```diff
@@ -57,4 +58,117 @@ class Hl7Ingest : CoolTest() {
+        val csv = """
+            A,b
+            1,2
+        """.trimIndent()
+        val a_60Meg_payload = "*".repeat(60000000)
+        val special_char = "!@#\\\\\\\$*()-_=+[]\\\\\\\\{};':,./<>?"
+        val xml_data = """
+            <?xml version="1.0"?>
+            <Tests xmlns="http://www.adatum.com">
+              <Test TestId="0001" TestType="CMD">
+                <Name>Convert number to string</Name>
+                <CommandLine>Examp1.EXE</CommandLine>
+                <Input>1</Input>
+                <Output>One</Output>
+              </Test>
+            </Tests>
+        """.trimIndent()
+        val badHl7Pairs = listOf(
+            /* Missing Required Fields */
+            Pair("MISSING PATIENT LASTNAME", cleanedHl7Message.replace("Patlast", "")),
+            Pair("MISSING ORDERING FACILITY STATE", cleanedHl7Message.replace("^IG^", "")),
+            Pair("MISSING MESSAGE ID", cleanedHl7Message.replace("2.5.1", "")),
+            Pair("MISSING TESTING LAB CLIA", cleanedHl7Message.replace("10D08761999", "")),
+            Pair("MISSING PATIENT STATE", cleanedHl7Message.replace("^MD^", "^^")),
+            /* Invalid data type */
+            Pair("INVALID DOB", cleanedHl7Message.replace("19670202", "19")),
+            Pair("INVALID DOB", cleanedHl7Message.replace("19670202", special_char)),
+            /* Bad hl7 files */
+            Pair("EMPTY FILE", ""),
+            Pair("CSV FILE", csv),
+            Pair("XML FILE", xml_data),
+            Pair("JSON", "{\"alive\": true}"),
+            /* Bad data */
+            Pair("Duplicate PID", cleanedHl7Message + "PID|||||Richards^Mary||19340428|F|||||||||||||||||||\\nPID|||||||19700510105000|M|||||||||||||||||||"),
+            Pair("Only MSH", "MSH|^~\\&|CDC PRIME - Atlanta, Georgia (Dekalb)^2.16.840.1.114222.4.1.237821^ISO|Avante at Ormond Beach^10D0876999^CLIA|PRIME_DOH|Prime Data Hub|20210210170737||ORU^R01^ORU_R01|371784|P|2.5.1|||NE|NE|USA||||PHLabReportNoAck^ELR_Receiver^2.16.840.1.113883.9.11^ISO"),
+            Pair("Partially Terminated", "MSH|^~\\&"),
+            Pair("RANDOM TEXT", "foobar"),
+            Pair("NON-UTF", "®"),
+            Pair("EMOJI", "❤️ \uD83D\uDC94 \uD83D\uDC8C \uD83D\uDC95 \uD83D\uDC9E \uD83D\uDC93"),
+            Pair("LARGE POST BODY", a_60Meg_payload),
+        )
+
+        for (pair in badHl7Pairs) {
+            val (responseCode, jsonResponse) = HttpUtilities.postReportBytes(
+                environment,
+                (pair.second).toByteArray(),
+                sender,
+                options.key
+            )
+
+            TermUi.echo("ResponseCode to POST: $responseCode")
+            val tree = jacksonObjectMapper().readTree(jsonResponse)
+            if (responseCode >= 400 && responseCode < 500 && tree["id"].isNull && tree["errorCount"].intValue() == tree["errors"].size() ) {
+                good("Test of Bad HL7 file ${pair.first} passed: Failure HttpStatus code was returned.")
```

Comment:
This message is not true.  You are testing more than httpstatus

Reply:
fixed